### ISSUE TITLE
Virtual 8 way dpad [enhancement]

### ISSUE THREAD

**Issue by Nicat93** [nikita36078/J2ME-Loader#329]
Which can be toggled on/off and is mapped to arrow keys. Something like this but transparent: 
![image](https://user-images.githubusercontent.com/8430147/39836346-10976dc8-53e4-11e8-8686-3b4a7e995c13.png) (pic from google)

**Comment by AdityaMH** [comment]
Adding virtual analog is great idea, but most j2me game using 4 d-way or numeric-pad.

**Comment by Nicat93** [comment]
@AdityaMH yes that is right. But J2ME already has diagonal direction support as arrow keys, so it shouldn't be a problem, right? 
If that is not the case then having ability to map directions to certain keys would be an ideal option. For example, Deep 3D uses 1,3,7,9 keys for diagonal movement which could be mapped to diagonal directions of dPad manually.

**Comment by AdityaMH** [comment]
yeah, you're right.. for Deep 3d and GoF, that very comfort using that. But  you know, virtual analog need lot screen space. If they phone only have small screen, that can be annoying too.

**Comment by Nicat93** [comment]
I don't know why you are thinking that way. If we had an analog we could just hide arrow/number keys which are used for directional movement. It would take same or even less space depending on design. Hopefully dev(s) will let us have it as an option in the future :)

**Comment by oatmeal01** [comment]
Dpad is nice for directional purpose but it takes alot of space in portrait. Aditya is right this could be a problem for small screen devices.
